pynestml/codegeneration/nest_codegenerator.py

```diff
@@ -75,51 +82,105 @@ class NESTCodeGenerator(CodeGenerator):
     - **simplify_expression**: For all expressions ``expr`` that are rewritten by ODE-toolbox: the contents of this parameter string are ``eval()``ed in Python to obtain the final output expression. Override for custom expression simplification steps. Example: ``sympy.simplify(expr)``. Default: ``"sympy.logcombine(sympy.powsimp(sympy.expand(expr)))"``. (This parameter is passed to ODE-toolbox.)
+    - **templates**: Path containing jinja templates used to generate code for NEST simulator.
+        - **path**: Path containing jinja templates used to generate code for NEST simulator.
+        - **model_templates**: A list of the jinja templates or a relative path to a directory containing the templates related to the neuron model(s).
+        - **module_templates**: A list of the jinja templates or a relative path to a directory containing the templates related to generating the NEST module.
     """
 
     _default_options = {
         "neuron_parent_class": "ArchivingNode",
         "neuron_parent_class_include": "archiving_node.h",
         "preserve_expressions": False,
-        "simplify_expression": "sympy.logcombine(sympy.powsimp(sympy.expand(expr)))"
+        "simplify_expression": "sympy.logcombine(sympy.powsimp(sympy.expand(expr)))",
+        "templates": {
+            "path": os.path.join(os.path.dirname(__file__), 'resources_nest', 'point_neuron'),
+            "model_templates": ['NeuronClass.cpp.jinja2', 'NeuronHeader.h.jinja2'],
+            "module_templates": ['setup']
+        }
     }
 
     _variable_matching_template = r'(\b)({})(\b)'
+    _model_templates = list()
+    _module_templates = list()
 
-    def __init__(self, options: Optional[Mapping[str, Any]]=None):
+    def __init__(self, options: Optional[Mapping[str, Any]] = None):
         super().__init__("NEST", options)
+        self._printer = ExpressionsPrettyPrinter()
         self.analytic_solver = {}
         self.numeric_solver = {}
-        self.non_equations_state_variables = {}   # those state variables not defined as an ODE in the equations block
-        self._setup_template_env()
+        self.non_equations_state_variables = {}  # those state variables not defined as an ODE in the equations block
+        self.setup_template_env()
+
+    def raise_helper(self, msg):
+        raise TemplateRuntimeError(msg)
 
-    def _setup_template_env(self):
-        """setup the Jinja2 template environment"""
+    def setup_template_env(self):
+        """
+        Setup the Jinja2 template environment
+        """
 
-        def raise_helper(msg):
-            raise TemplateRuntimeError(msg)
+        # Get templates path
+        templates_root_dir = self.get_option("templates")['path']
+        if not os.path.isabs(templates_root_dir):
+            raise InvalidPathException('Templates path '"" + templates_root_dir + ""' is not an absolute path.')
```

Comment:
What do you think about using the default location (``pynestml/codegeneration/resources_nest``) if a relative path is provided?

Reply:
This is a good idea. I have therefore prefixed the default templates path if the absolute path is not provided. Thanks!